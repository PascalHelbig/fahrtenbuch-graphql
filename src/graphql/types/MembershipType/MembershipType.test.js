const { makeExecutableSchema, addMockFunctionsToSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const MembershipType = require('./MembershipType');

it('has to be valid schema', () => {
  const SchemaDefinition = 'schema { query: Membership }';
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, MembershipType] });
  addMockFunctionsToSchema({
    schema,
    mocks: {
      Membership: () => ({
        user: () => ({ id: 1337 }),
        group: () => ({ id: 1338 }),
        is_admin: () => true,
      }),
    },
  });

  const query = `{
    user { id }
    group { id }
    is_admin
  }`;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot()
  );
});
