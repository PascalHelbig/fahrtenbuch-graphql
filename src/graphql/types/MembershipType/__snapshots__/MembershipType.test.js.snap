exports[`test has to be valid schema 1`] = `
Object {
  "data": Object {
    "group": Object {
      "id": "1338",
    },
    "is_admin": true,
    "user": Object {
      "id": "1337",
    },
  },
}
`;
